crates/re_viewer_context/src/selection_state.rs

```diff
@@ -350,11 +355,27 @@ impl ApplicationSelectionState {
                                 &current_instance_path.instance_key.specific_index(),
                                 &test_instance_path.instance_key.specific_index(),
                             )
-                            && either_none_or_same(current_space_view_id, test_space_view_id)
-                    } else {
-                        false
                     }
-                }
+                },
+
+                Item::DataResult(_current_space_view_id, current_instance_path) => match test {
+                    Item::StoreId(_)
+                    | Item::ComponentPath(_)
+                    | Item::SpaceView(_)
+                    | Item::Container(_) => false,
+                    Item::InstancePath(test_instance_path)
+                    | Item::DataResult(_, test_instance_path) => {
+                        fn either_none_or_same<T: PartialEq>(a: &Option<T>, b: &Option<T>) -> bool {
+                            a.is_none() || b.is_none() || a == b
```

Comment:
micro-nit: you're using this several times in the same file, could move it out (and who knows maybe it's used even more often; seems like a nice utility; might even consider putting it on an extension trait)